v03_pipeline/lib/misc/lookup.py

```diff
@@ -0,0 +1,235 @@
+import hail as hl
+
+from v03_pipeline.lib.model import DatasetType
+
+
+def compute_callset_lookup_ht(
+    dataset_type: DatasetType,
+    mt: hl.MatrixTable,
+    project_guid: str,
+) -> hl.Table:
+    sample_id_to_family_guid = hl.dict(
+        {
+            s: family_guid
+            for family_guid, sample_ids in hl.eval(mt.family_samples).items()
+            for s in sample_ids
+        },
+    )
+    ht = mt.select_rows(
+        project_stats=hl.array(
+            [
+                hl.agg.collect(
+                    hl.Struct(
+                        family_guid=sample_id_to_family_guid[mt.s],
+                        **mt.entry,
+                    ),
+                )
+                .group_by(lambda fs: fs.family_guid)
+                .values()
+                .map(
+                    lambda fs: hl.Struct(
+                        family_guid=fs[0].family_guid,
+                        **{
+                            field_name: hl.len(fs.filter(filter_fn))
+                            for field_name, filter_fn in dataset_type.lookup_table_fields_and_genotype_filter_fns.items()
+                        },
+                    ),
+                ),
+            ],
+        ),
+    ).rows()
+    return globalize_ids(ht, project_guid)
+
+
+def globalize_ids(ht: hl.Table, project_guid: str) -> hl.Table:
+    row = ht.take(1)
+    has_project_stats = row and len(row[0].project_stats) > 0
+    ht = ht.annotate_globals(
+        project_guids=[project_guid],
+        project_families=(
+            {project_guid: [fs.family_guid for fs in ps] for ps in row[0].project_stats}
+            if has_project_stats
+            else hl.empty_dict(hl.tstr, hl.tarray(hl.tstr))
+        ),
+    )
+    return ht.annotate(
+        project_stats=ht.project_stats.map(
+            lambda ps: ps.map(lambda fs: fs.drop('family_guid')),
+        ),
+    )
+
+
+def remove_new_callset_family_guids(
+    ht: hl.Table,
+    project_guid: str,
+    family_guids: list[str],
+) -> hl.Table:
+    if project_guid not in hl.eval(ht.globals.project_families):
+        return ht
+    family_guids = hl.set(family_guids)
+    project_i = ht.project_guids.index(project_guid)
+    family_indexes_to_keep = hl.array(
+        hl.enumerate(ht.globals.project_families[project_guid])
+        .filter(lambda item: ~family_guids.contains(item[1]))
+        .map(lambda item: item[0]),
```

Comment:
this logic is shared with 'family_entries.py', maybe it belongs in a helper function

Reply:
also why do we need to do this? why are there extra families inside of a 1project_families` project?

Reply:
The field name is `family_guids` there rather than `ht.globals.project_families[project_guid]`, though you're right that the rest of the logic is the same.  There's an additional nested layer here because the lookup table contains all projects rather than just one.  

The idea here is that we want to remove families from the table if they've been re-called and are present in the new callset.

Reply:
its out of scope for this PR but at some point in the future we will want to add the ability to just remove families from seqr without reloading them and we will probably use this helper function. Since theres nothing in this function thats specifically related to the new callset, might be more maintainable to just name it `remove_family_guids` or something like that